### PR TITLE
Add test for multiple inference rules

### PR DESCRIPTION
**Pull request by kpanic**
hey there 👋🏾 

I was wondering if we can have multiple inference in a chain.

I am providing a test to show what I mean.

thanks!

### CODE REVIEW

File: test/engine_test.exs [hunk 1]
```diff
@@ -9,6 +9,89 @@ defmodule NeuralBridge.EngineTest do
...
+
+    engine =
+      Engine.add_facts(engine, """
+      Patient's fever is 39
+      Patient's name is "Aylon"
+      Patient's generic_weakness is "Yes"
+      """)
+
+    engine =
+      Enum.reduce(engine.rule_engine.agenda, engine, fn pnode, network ->
```
Comment (lorenzosinisi) [inline test/engine_test.exs]: This will work if we consume all the rules until none will apply (or has been applied already) 👍 no?

Comment (lorenzosinisi) [reply test/engine_test.exs]: In other words: 

1. call apply_rule -> store the rule ID applied 
2. check the agenda and apply rules again -> store new rule applied IDs
3. check the agenda and apply rules again -> store the new rule applied IDs
4. there is no difference between the rules in agenda and the rules applied -> stop the cycle

Comment (kpanic) [reply test/engine_test.exs]: @lorenzosinisi store where? Thanks for the full rundown!

Comment (lorenzosinisi) [reply test/engine_test.exs]: Store it in a variable or in memory, I have an example implementation somewhere wait: 

```elixir
  def add_facts(session = %__MODULE__{rule_engine: rule_engine}, facts) when is_binary(facts) do
    ast_facts = NeuralBridge.SanskritInterpreter.to_production!(facts)
    new_rule_engine = Enum.reduce(ast_facts, rule_engine, &Retex.add_wme(&2, &1))
    new_session = %{session | rule_engine: new_rule_engine}
    apply_rules(new_session)
  end

  def add_facts(session = %__MODULE__{rule_engine: rule_engine}, facts) when is_list(facts) do
    new_rule_engine = Enum.reduce(facts, rule_engine, &Retex.add_wme(&2, &1))
    new_session = %{session | rule_engine: new_rule_engine}

    apply_rules(new_session)
  end

  @doc "Take the list of applicable rules from the agenda and trigger them if they haven't been applied yet"
  @spec apply_rules(t()) :: t()
  def apply_rules(session = %__MODULE__{rule_engine: rule_engine, rules_fired: rules_fired}) do
    rules = extract_applicable_rules(rule_engine, rules_fired)
    updated_session = Enum.reduce(rules, session, &apply_rule(&2, &1))
    if Enum.any?(rules), do: apply_rules(updated_session), else: updated_session
  end

  defp extract_applicable_rules(%_{agenda: agenda}, rules_fired) do
    agenda -- rules_fired
  end
  ```

Comment (kpanic) [reply test/engine_test.exs]: Thanks to the latest changes, the test passes! 🎉 
In this PR, after rebase, there is one test that was kept.
It was removed intentionally or should I remove it @lorenzosinisi ?

Comment (lorenzosinisi) [reply test/engine_test.exs]: It was removed intentionally as this new version is almost a complete rewrite and much more robust than the previous one, will keep adding more tests soon but feel free to open new issues if you find something else!

Now the engine will also error and provide error messages on syntax issues or invalid syntax statements in the "given" 😎

It should work just fine ™️ but there is a lot that can be improved: loop detection would be one thing, battle test the supported functions and allow it to be configurable (so one can use neural_engine as a library and use custom functions) and more

Comment (kpanic) [reply test/engine_test.exs]: Noice. Makes sense. I have just removed the old test!

**Pull request merged**